plugins/block-dynamic-connection/src/dynamic_list_create.ts

```diff
@@ -125,7 +125,7 @@ const DYNAMIC_LIST_CREATE_MIXIN = {
    */
   loadExtraState: function (
     this: DynamicListCreateBlock,
-    state: {[x: string]: any} | string,
+    state: {[x: string]: number} | string,
```

Comment:
I don't think this is quite right for the `state` types, here and below. The extraState could have any number of properties on it (we don't know) and it's not correct to assume all of those properties will be numbers (or strings, or whatever). We don't know the types. To be strict, we should type that as `unknown` rather than `any`.

I think the *most* correct would be

```
state: { itemCount?: number, [x: string]: unknown} | string,
```

This says the state can either be a string (when it's xml), or it can be an object with any number of properties whose type is `unknown`, and also it might have a property called `itemCount` that's a number. Then you'll get an error later because `unknown` is a stricter type than `any`. You'll need to change `this.itemCount = state['itemCount'];` to `this.itemCount = state['itemCount'] ?? 0;` to properly account for the case where `itemCount` does not exist in the extraState.

This might seem like a lot of work, but now the code will be truly type-safe instead of the current state which uses `any` to basically turn off type checking.

You would need a similar fix for the others below, but instead of `itemCount` it would be whatever property is actually used in the body of the function.